tsconfig.json

```diff
@@ -1,12 +1,24 @@
 {
   "compilerOptions": {
     "lib": ["es5", "es6", "dom"],
-    "noEmitOnError": true,
-    "noImplicitAny": true,
+    "downlevelIteration": true,
+    "moduleResolution": "node",
+    "experimentalDecorators": true,
+    "emitDecoratorMetadata": true,
+    "skipLibCheck": true,
     "noImplicitThis": true,
+    "noFallthroughCasesInSwitch": true,
     "noUnusedLocals": true,
     "noUnusedParameters": true,
+    "noErrorTruncation": true,
+    "noEmitOnError": false,
+    "importHelpers": true,
+    "noStrictGenericChecks": true,
+    "strictNullChecks": true,
     "alwaysStrict": true,
+    "strict": true,
```

Comment:
We don't need `strictNullChecks`, `alwaysStrict`, `strictPropertyInitialization`, and `strictFunctionTypes` when `strict` is true, as it already includes them. You can remove those others.